dev/make-distribution.sh

```diff
@@ -286,6 +286,6 @@ if [ "$MAKE_TGZ" == "true" ]; then
   TARDIR="$SPARK_HOME/$TARDIR_NAME"
   rm -rf "$TARDIR"
   cp -r "$DISTDIR" "$TARDIR"
-  tar czf "spark-$VERSION-bin-$NAME.tgz" -C "$SPARK_HOME" "$TARDIR_NAME"
```

Comment:
Ur, @koertkuipers . It seems that it's not robust enough. I'm using the latest `macOS Mojave 10.14.4`, but it fails by default. Although we are using `spark-rm` docker image, Mac is important dev environment and this failure is a regression.

```
$ tar --version
bsdtar 2.8.3 - libarchive 2.8.3

$ tar czf "test.tgz" --numeric-owner --owner=0 "docs"
tar: Option --owner=0 is not supported
Usage:
  List:    tar -tf <archive-filename>
  Extract: tar -xf <archive-filename>
  Create:  tar -cf <archive-filename> [filenames...]
  Help:    tar --help
```

Reply:
Ah, this is a gnu vs BSD tar thing, darn. I don't see any equivalent on OS X and yeah we don't want to make this fail on Macs.
Can you use -o when extracting the archive as root to avoid this instead?

Reply:
> Ah, this is a gnu vs BSD tar thing, darn. I don't see any equivalent on OS X and yeah we don't want to make this fail on Macs.
> Can you use -o when extracting the archive as root to avoid this instead?

i think you can use -o yes to extract it. however quite a few tools dont do this, for example the default chef resource to deploy tarballs (ark, https://github.com/chef-cookbooks/ark) does not. this is how i ran into this. i noticed files being deployed with my personal uid/gid on servers because i was the one that ran `dev/make-distribution.sh`. the standard for tarballs seems to be to have uid/gid set to 0.

anyhow seems there isnt much we can do. we can close this.